pkgs/by-name/ap/apache-orc/package.nix

```diff
@@ -44,12 +44,19 @@ stdenv.mkDerivation (finalAttrs: {
     zstd
   ];
 
-  cmakeFlags = [
-    (lib.cmakeFeature "CMAKE_BUILD_TYPE" "Release")
-    (lib.cmakeBool "BUILD_JAVA" false)
-    (lib.cmakeBool "STOP_BUILD_ON_WARNING" true)
-    (lib.cmakeBool "INSTALL_VENDORED_LIBS" false)
-  ];
+  cmakeFlags =
+    [
+      (lib.cmakeFeature "CMAKE_BUILD_TYPE" "Release")
+      (lib.cmakeBool "BUILD_JAVA" false)
+      (lib.cmakeBool "STOP_BUILD_ON_WARNING" true)
+      (lib.cmakeBool "INSTALL_VENDORED_LIBS" false)
+    ]
+    ++ lib.optional (stdenv.hostPlatform != stdenv.buildPlatform) [
+      # Fix (RiscV) cross-compilation
+      (lib.cmakeFeature "HAS_PRE_1970_EXITCODE" "0")
+      (lib.cmakeFeature "HAS_POST_2038_EXITCODE" "0")
```

Comment:
Maybe we should set those based on this? https://github.com/apache/orc/blob/09693fee2d65a49a6a65b5774cf4eb2001bedf3c/conan/all/conanfile.py#L161-L162

no idea if that would be correct